File: Arrays/secondLargestElement/secondLargestOptimal.js
let arr = [1,2,4,5,6,23,5,11,33,32];
let largestNum = arr[0];
let secondLargest = -1;

for(let i = 0; i <arr.length; i++){
    if(arr[i] > largestNum){
        secondLargest = largestNum;
        largestNum = arr[i];
    }
    else if(arr[i] < largestNum && arr[i] > secondLargest){
        secondLargest = arr[i];
    }
}

console.log(secondLargest);

File: Arrays/secondLargestElement/secondLargestOptimal.cpp
#include<iostream>
using namespace std;
int main(){
    int n = 1;
    int arr[] = {1};

    int largest = arr[0];
    int secondLargest = -1;

    for(int i = 0; i < n;i++){
        if(arr[i] > largest){
            secondLargest = largest;
            largest = arr[i];
        }
        else if (arr[i] < largest && arr[i] > secondLargest){
            secondLargest = arr[i];
        }
    }

    cout << "Second Largest :" << secondLargest;
}